src/core/layout/qgslayoutitemmap.cpp

```diff
@@ -2226,7 +2232,12 @@ void QgsLayoutItemMap::refreshMapExtents( const QgsExpressionContext *context )
   QgsExpressionContext scopedContext;
   if ( !context )
     scopedContext = createExpressionContext();
+
+  bool ok = false;
   const QgsExpressionContext *evalContext = context ? context : &scopedContext;
+  QString crsVar = mDataDefinedProperties.valueAsString( QgsLayoutObject::MapCRS, *evalContext, QString(), &ok );
+  if ( ok )
+    setCrs( crsVar );
```

Comment:
let's drop the new public `setCrs( const QString &crs ) ` method, and just put that logic in here instead. (Avoids adding complexity to the public api, adding new methods to the public stable api)

Reply:
Sure thing.